### ISSUE TITLE
Linking empty rust file fails with "unable to find library -lc-debug"

### ISSUE THREAD

**Issue by hoodmane** [emscripten-core/emscripten#17191]
The following:
```sh
touch file.rs
rustc file.rs --target wasm32-unknown-emscripten -C link-arg=-sSIDE_MODULE=1 --crate-type cdylib 
```
Results in
`wasm-ld: error: unable to find library -lc-debug`.
Rust always includes `-sASSERTIONS=1` in the linker args, see https://github.com/rust-lang/rust/pull/97928/.

This seems to be a regression between emscripten version 2.0.27 and version 3.1.6. If I surgically set `settings.ASSERTIONS=0`, then it still fails with "unable to find library -lc". I don't think it should be trying to link `-lc` anyways because I am trying to make a side module. But I am not sure what the proper setting is to convince Rust or Emscripten to discard this.

**Comment by sbc100** [comment]
Can you show the failing wasm-ld comment?  

I'm pretty sure emscripten will never add `-lc` or `-lc-debug` when linking a side module, so I can only imagine that the rust driver is (erroneously) adding it.  

I think you best bet is to remove any/all explicit additions of `-lc`, since the compiler driver (emcc) will add it if/when its needed.

(I don't think this is actually and emscripten bug BTW, but a rustc driver bug)

**Comment by hoodmane** [comment]
Yes Rust is adding it. I agree it's probably not an emscripten bug. It's added by Rust in a way that makes it hard to remove, but if I patch emcc to delete it I am getting the following:
```
parse exception: attempted pop from empty stack / beyond block start boundary at 268272 (at 0:268272)]
          Fatal: error in parsing input
          emcc: error: emcc: error: '/home/hood/Documents/programming/emsdk/upstream/bin/wasm-emscripten-finalize --dyncalls-i64 --pass-arg=legalize-js-interface-export-originals --side-module main.wasm -o main.wasm --detect-features'
```

**Comment by sbc100** [comment]
That looks like an invalid wasm file... can you gram `main.wasm`.. does it pass `wasm-validate` ?

**Comment by hoodmane** [comment]
Returns a lot of errors like:
```
main.wasm:00417f0: error: type mismatch in call, expected [i32, i32, i32, i32] but got [i32, i32, i32]
main.wasm:00417f1: error: type mismatch in drop, expected [any] but got []
main.wasm:0041b3f: error: type mismatch in function, expected [] but got [i32, i64]
main.wasm:004217b: error: type mismatch in function, expected [] but got [i32]
main.wasm:00422e4: error: type mismatch in function, expected [] but got [i32]
...
```

**Comment by hoodmane** [comment]
`-C lto=thin` seems to fix it.

**Comment by sbc100** [comment]
Can you see which function is at those offsets?  (`wasm-objdump -D` and then search for the hex offsets)
